src/aiida_quantumespresso/parsers/parse_raw/ph.py

```diff
@@ -471,4 +478,10 @@ def parse_initialization_qpoints(stdout: str) -> dict:
 
         parameters.update({'q_points': q_points})
 
-    return parameters
+    if 'number_of_qpoints' not in parameters and 'q_points' not in parameters:
+        return parameters, False
+
+    if parameters['number_of_qpoints'] != len(parameters['q_points']):
+        return parameters, False
+
+    return parameters, True
```

Comment:
I am not a big fan of communicating errors through return values, especially not if that means having to turn the return value into a tuple. Since you only really use the `parameters` result in case there isn't a problem, what is the problem with just raising an exception and catching that in the caller?